File: task3/task3.cpp
#include <cassert>
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n = 0;
    cin >> n;
    vector<string> numbers;
    for (int i = 0; i < n; ++i) {
        string number = "";
        cin >> number;
        for (auto s : number) {
            assert(isdigit(s));
        }
        numbers.push_back(number);
    }
    for (auto s : numbers) {
        int counter = 0;
        if (s.length() == 1) {
            cout << "-" << " ";
            continue;
        }
        for (auto t : s) {
            if (static_cast<int>(t) % 2 == 0) {
                ++counter;
            }
        }
        cout << counter << " ";
    }
    return 0;
}

File: task3/task3.swift
import Foundation

guard let firstLine = readLine(), let n = Int(firstLine) else {
    fatalError("Некорректный ввод для n")
}

var numbers: [String] = []

for _ in 0..<n {
    guard let number = readLine() else {
        fatalError("Недостаточно входных данных")
    }
    for ch in number {
        assert(ch.isNumber, "Символ должен быть цифрой")
    }
    numbers.append(number)
}

var results: [String] = []

for s in numbers {
    if s.count == 1 {
        results.append("-")
        continue
    }
    var counter = 0
    for ch in s {
        if let ascii = ch.asciiValue, ascii % 2 == 0 {
            counter += 1
        }
    }
    results.append(String(counter))
}

print(results.joined(separator: " "))